File: IT/L'Indirizzario/Java/Indirizzario Java v1/src/MainFrame.java
/* file MainFrame.java */

package programmiamo.ind;

import java.awt.Dimension;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import javax.swing.JFrame;
import javax.swing.JMenuBar;
import javax.swing.JMenu;
import javax.swing.JMenuItem;
import javax.swing.JTabbedPane;
import javax.swing.JPanel;

public class MainFrame extends JFrame {
	private JTabbedPane tabs;

	public MainFrame() {
		/* crea il Notebook */

		tabs = new JTabbedPane();

		updateUI();

		JMenuBar menuBar = new JMenuBar();
		JMenu fileMenu = new JMenu("File");
		menuBar.add(fileMenu);

		/* aggiunge tre sottomenu al menu "file" */

		JMenuItem save = new JMenuItem("Nuovo");
		JMenuItem remove = new JMenuItem("Cancella");
		JMenuItem exit = new JMenuItem("Esci");

		fileMenu.add(save);
		fileMenu.add(remove);
		fileMenu.add(exit);

		/* gestisce il click sul menu "Nuovo" */

		save.addActionListener(	new ActionListener() {
			public void actionPerformed(ActionEvent evt) {
				;
			}
		});

		/* gestisce il click sul menu "Cancella" */ 

		remove.addActionListener(	new ActionListener() {
			public void actionPerformed(ActionEvent e) {
				;
			}
		});

		/* gestisce il click sul menu "Esci" */ 

		exit.addActionListener(	new ActionListener() {
			public void actionPerformed(ActionEvent e) {
				MainFrame.this.dispose();
			}
		});

		this.setJMenuBar(menuBar);
		this.add(tabs);
		this.setSize(new Dimension(640, 480));
		this.setTitle("Indirizzario");
		this.setVisible(true);
		this.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
	}

	/* crea le linguette nel Notebook */

	public void updateUI() {
		for (int counter = 1; counter < 4; counter++) {
			String pageName = "Pagina " + counter;
			tabs.addTab(pageName, new JPanel());
		}
	}
}

File: IT/2/L'Indirizzario/C++/Indirizzario C++ v1/MainFrame.cpp
/* file MainFrame.cpp */

#include <QApplication>
#include <QMenuBar>
#include <QVBoxLayout>
#include "MainFrame.h"

namespace ind = programmiamo::ind;

ind::MainFrame::MainFrame(QWidget *parent) : QMainWindow(parent) 
{
	/* crea il Notebook */

	tabs = new QTabWidget(this);

	updateUI();

	QMenu *fileMenu = menuBar()->addMenu("&File");

	/* aggiunge tre sottomenu al menu "file" */

	QAction *saveAction = new QAction("Nuovo", this);
	QAction *removeAction = new QAction("Cancella", this);
	QAction *exitAction = new QAction("Esci", this);

	fileMenu->addAction(saveAction);
	fileMenu->addAction(removeAction);
	fileMenu->addAction(exitAction);

	/* associa a ogni voce del menu il relativo gestore di evento */

	connect(saveAction, SIGNAL(triggered()), this, SLOT(save()));
	connect(removeAction, SIGNAL(triggered()), this, SLOT(remove()));
	connect(exitAction, SIGNAL(triggered()), this, SLOT(exit()));

	QWidget *panel = new QWidget(this);
	this->setCentralWidget(panel);

	QVBoxLayout *box = new QVBoxLayout();
	box->setMargin(0);
	box->addWidget(tabs);
	panel->setLayout(box);

	// 600 invece di 640?
	this->resize(640, 480);
	this->show();
}

/* gestisce il click sul menu "Nuovo": per ora non fa nulla */

void ind::MainFrame::save() {}

/* gestisce il click sul menu "Cancella": per ora non fa nulla */ 

void ind::MainFrame::remove() {}

/* gestisce il click sul menu "Esci", chiudendo l'applicazione */ 

void ind::MainFrame::exit()
{
	QApplication::quit();
}

/* crea le linguette nel Notebook */

void ind::MainFrame::updateUI()
{
	for (int counter = 1; counter < 4; counter++) {
		QString pageName = "Pagina " + QString::number(counter);
		tabs->addTab(new QWidget(), pageName);
	}
}

